### DISCUSSION TITLE
how to download build output from Vercel for scanning Fortify and Blackduck?

### DISCUSSION THREAD

**Discussion by ashokvaddevalli** [vercel/community#6607] [Help]
### Summary

We are using github actions for build and deploy project on Vercel and builds, deployment happening on vercel platform. But we need compiled source files and packages (build workspace) for Fortify and Blackduck scan.  

### Steps to Reproduce

We are tried download build output from Vercel project but didn't findout download option. Is there option to download build output folder.

**Comment by amyegan** [comment]
Hey @ashokvaddevalli. The easiest way is to generate the output locally using [Vercel CLI](https://vercel.com/docs/cli). You would use [`vercel build`](https://vercel.com/docs/cli/build) or [`vercel build --prod`](https://vercel.com/docs/cli/build#production) to run a local build which should result in the same output you would see on the server.

There are also a couple of Vercel API endpoints to [list deployment files](https://vercel.com/docs/rest-api/endpoints/deployments#list-deployment-files) and [get deployment file contents](https://vercel.com/docs/rest-api/endpoints/deployments#get-deployment-file-contents) which may be of use to you.

**Reply by ashokvaddevalli** [reply]
Hi @amyegan  I think build happen on Vercel platform yes using Vercel build and not local machine yes?

**Reply by ashokvaddevalli** [reply]
Hi @amyegan 
Here is the githubflow I am using

name: GitHub Actions Vercel Develop Deployment
env:
  
  VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
  VERCEL_PROJECT_ID: ${{ secrets.VERCEL_PROJECT_ID }}
on:  
    push:
     branches:
      - main
    
jobs:
  Deploy-Preview:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3

      - name: Install Node.js
        uses: actions/setup-node@v3
        with:
          node-version: 18

      - name: Install Vercel CLI
        run: npm install --location=global vercel@canary pnpm@8.15.7

      - name: Pull Vercel Environment Information
        run: vercel pull --yes --environment=preview --token=${{ secrets.VERCEL_TOKEN }}

      - name: Build Project Artifacts
        run: vercel build --yes --token=${{ secrets.VERCEL_TOKEN }}

      - name: Deploy Project Artifacts to Vercel
        run: |
          vercel deploy --prebuilt --token=${{ secrets.VERCEL_TOKEN }} > domain.txt
          vercel alias --token=${{ secrets.VERCEL_TOKEN }} set `cat domain.txt` ashokv-iota.vercel.app
